docs/single-to-multi-master.md

```diff
@@ -0,0 +1,213 @@
+
+## Warnings
+
+This is a risky procedure that **can lead to data-loss** in the etcd cluster.
+Please follow all the backup steps before attempting it. Please read the
+[etcd admin guide](https://github.com/coreos/etcd/blob/v2.2.1/Documentation/admin_guide.md)
+before attempting it.
+
+During this procedure, you will experience **downtime** on the API server, but
+not on the end user services.
+
+## 1 - Backups
+
+### a - Backup main etcd cluster
+
+```bash
+$ kubectl --namespace=kube-system get pods | grep etcd
+etcd-server-events-ip-172-20-36-161.ec2.internal        1/1       Running   4          2h
+etcd-server-ip-172-20-36-161.ec2.internal               1/1       Running   4          2h
+$ kubectl --namespace=kube-system exec etcd-server-ip-172-20-36-161.ec2.internal -it -- sh
+/ # etcdctl backup --data-dir /var/etcd/data --backup-dir /var/etcd/backup
+/ # mv /var/etcd/backup/ /var/etcd/data/
+/ # exit
+$ kubectl --namespace=kube-system get pod etcd-server-ip-172-20-36-161.ec2.internal -o json | jq '.spec.volumes[] | select(.name | contains("varetcdata")) | .hostPath.path'
+"/mnt/master-vol-0ea119c15602cbb57/var/etcd/data"
+$ ssh admin@<master-node>
+admin@ip-172-20-36-161:~$ sudo -i
+root@ip-172-20-36-161:~# mv /mnt/master-vol-0ea119c15602cbb57/var/etcd/data/backup /home/admin/
+root@ip-172-20-36-161:~# chown -R admin: /home/admin/backup/
+root@ip-172-20-36-161:~# exit
+admin@ip-172-20-36-161:~$ exit
+$ scp -r admin@<master-node>:backup/ .
+```
+
+### b - Backup event etcd cluster
+
+```bash
+$ kubectl --namespace=kube-system exec etcd-server-events-ip-172-20-36-161.ec2.internal -it -- sh
+/ # etcdctl backup --data-dir /var/etcd/data-events --backup-dir /var/etcd/backup
+/ # mv /var/etcd/backup/ /var/etcd/data-events/
+/ # exit
+$ kubectl --namespace=kube-system get pod etcd-server-events-ip-172-20-36-161.ec2.internal -o json | jq '.spec.volumes[] | select(.name | contains("varetcdata")) | .hostPath.path'
+"/mnt/master-vol-0bb5ad222911c6777/var/etcd/data-events"
+$ ssh admin@<master-node>
+admin@ip-172-20-36-161:~$ sudo -i
+root@ip-172-20-36-161:~# mv /mnt/master-vol-0bb5ad222911c6777/var/etcd/data-events/backup/ /home/admin/backup-events
+root@ip-172-20-36-161:~# chown -R admin: /home/admin/backup-events/
+root@ip-172-20-36-161:~# exit
+admin@ip-172-20-36-161:~$ exit
+$ scp -r admin@<master-node>:backup-events/ .
+```
+
+## 2 - Add a new master
+
+### a - Create the instance group
+
+Create 1 kops instance group for the first one of your new masters, in
+a different AZ from the existing one.
+
+```bash
+$ kops create instancegroup master-<availability-zone2>
+```
+
+ * ``maxSize`` and ``minSize`` should be 1,
+ * ``role`` should be ``Master``,
+ * only one zone should be listed.
+
+### b - Reference the new masters in your cluster configuration
+
+*kops will refuse to have only 2 members in the etcd clusters, so we have to
+reference a third one, even if we have not created it yet.*
+
+```bash
+$ kops edit cluster myclusterdomain.com
```

Comment:
We've started using `example.com` in our docs for the cluster domain. Please switch to using that wherever you are using `myclusterdomain.com`.